File: Data Structures & Algorithms/duplicate-integer/submission-3.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        std::unordered_set<int> x;
        for(int i(0); i < nums.size(); i++){
            if(x.contains(nums[i])){
                return true;
            }
            x.insert(nums[i]);
        }
        return false;
    }
};

File: Data Structures & Algorithms/two-integer-sum/submission-3.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map = {}
        for i in range(len(nums)):
            wi = target - nums[i]
            if wi in map:
                return [map.get(wi), i]
            map[nums[i]] = i
        return [-1, -1]